File: app/src/main/java/com/dagger/raju/myapplication/di/ReceiversModule.java
package com.dagger.raju.myapplication.di;

import android.support.annotation.NonNull;

import com.dagger.raju.myapplication.network.NetworkChannel;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by raju on 7/4/19.
 */

@Module
public class ReceiversModule {

    @Provides
    @NonNull
    @Singleton
    NetworkChannel providesNetworkChannel(){
        return new NetworkChannel();
    }
}

File: app/src/main/java/com/dagger/raju/myapplication/di/ReceiversModule.kt
package com.dagger.raju.myapplication.di

import com.dagger.raju.myapplication.network.NetworkChannel
import org.koin.dsl.module

/**
 * Created by raju on 7/4/19.
 */

val receiverModule = module {
    single { NetworkChannel() }
}